src/displayapp/screens/CatDog.cpp

```diff
@@ -0,0 +1,283 @@
+
+  UpdateLife();
+
+  taskRefresh = lv_task_create(RefreshTaskCallback, LV_DISP_DEF_REFR_PERIOD, LV_TASK_PRIO_MID, this);
+}
+
+CatDog::~CatDog() {
+  lv_task_del(taskRefresh);
+  lv_obj_clean(lv_scr_act());
+}
+
+void CatDog::Refresh() {
+  switch (state) {
+    case CatDog::State::Idle:
+      // add a delay to reject previous press event
+      if (animationTime > 0) {
+        animationTime--;
+      } else {
+        if (isGameStopped) {
+          state = CatDog::State::Stopped;
+        } else {
+          state = CatDog::State::Prompt;
+        }
+      }
+      break;
+    case CatDog::State::Stopped:
+      // start game
+      if (isPressing) {
+        lv_obj_align(windBarBG, lv_scr_act(), LV_ALIGN_IN_TOP_MID, 0, 30);
+        catLifePoint = 100;
+        dogLifePoint = 100;
+        UpdateLife();
+        isDogsTurn = true;
+        animationTime = 20;
+        isGameStopped = false;
+        state = CatDog::State::Idle;
+      }
+      break;
+    case CatDog::State::Prompt:
+      RandomWind();
+      if (isDogsTurn) {
+        lv_label_set_text_static(info, "Your turn");
+        lv_obj_align(info, lv_scr_act(), LV_ALIGN_IN_TOP_MID, 0, 40);
+        state = CatDog::State::WaitUser;
+      } else {
+        lv_label_set_text_static(info, "Cat's turn");
+        lv_obj_align(info, lv_scr_act(), LV_ALIGN_IN_TOP_MID, 0, 40);
+        lv_obj_align(powerBar, lv_scr_act(), LV_ALIGN_IN_TOP_MID, 0, 70);
+        // throwPower for perfect hit is 370 + wind*12;
+        PCTargetThrowPower = 343 + wind * 6 + rand() % 54;
+        lv_obj_set_pos(bone, 215, HORIZON / 100);
+        state = CatDog::State::ComputerThrowing;
+      }
+      break;
+    case CatDog::State::WaitUser:
+      if (isPressing) {
+        lv_obj_align(powerBar, lv_scr_act(), LV_ALIGN_IN_TOP_MID, 0, 70); // show powerBar
+        lv_obj_set_pos(bone, 5, HORIZON / 100);
+        state = CatDog::State::UserThrowing;
+      }
+      break;
+    case CatDog::State::UserThrowing:
+      if ((!isPressing && throwPower > MIN_POWER + 20) || throwPower >= MAX_POWER) {
+        Throw(throwPower);
+      } else {
+        throwPower += 3;
+        lv_bar_set_value(powerBar, throwPower, LV_ANIM_OFF);
+      }
+      break;
+    case CatDog::State::ComputerThrowing:
+      if (throwPower < PCTargetThrowPower) {
+        throwPower += 3;
+        lv_bar_set_value(powerBar, throwPower, LV_ANIM_OFF);
+      } else {
+        Throw(-throwPower);
+      }
+      break;
+    case CatDog::State::BoneFlying:
+      boneVY += 33;   // gravity
+      boneVX += wind; // acceleration caused by wind
+      boneX += boneVX;
+      boneY += boneVY;
+      lv_obj_set_pos(bone, boneX / 100, boneY / 100);
+      if (boneY >= HORIZON || boneX > 24000 || boneX < -2400) {
+        if (-2400 < boneX && boneX < 24000) {
+          if (isDogsTurn) {
+            handleDamage(abs(boneX - 21500) / 100);
+          } else {
+            handleDamage(abs(boneX - 500) / 100);
```

Comment:
Where are all those magic numbers coming from? Can we replace them by a variable with a good name and `constexpr`?